crypto/stack/stack.c

```diff
@@ -173,9 +173,10 @@ static int sk_reserve(OPENSSL_STACK *st, int n, int exact)
          * At this point, |st->num_alloc| and |st->num| are 0;
          * so |num_alloc| value is |n| or |min_nodes| if greater than |n|.
          */
-        st->data = OPENSSL_zalloc(sizeof(void *) * num_alloc);
-        if (st->data == NULL)
+        if ((st->data = OPENSSL_zalloc(sizeof(void *) * num_alloc)) == NULL) {
+            /* STACKerr(STACK_F_SK_RESERVE, ERR_R_MALLOC_FAILURE); */
```

Comment:
Errors from stack are reported by caller and stack user...

Reply:
I don't understand this comment?

Reply:
I say that this code is useless: 
it has no value added, it just says that one of  the many stack is KO, ... and please guess which one  ;) 
Actually, the current usage is to report this kind of failure just from the caller code, just after the failed call .

Reply:
No, fine-grain "what went wrong" and then pushing more errors as we pop up the stack is the  way we do things.